.github/workflows/test-prs.yml

```diff
@@ -133,16 +171,29 @@ jobs:
 
       - name: Publish Test Report
         uses: mikepenz/action-junit-report@v4
-        if: success() || failure() # always run even if the previous step fails
+        if: always() # always run even if the previous step fails
         with:
           report_paths: '**/*.xml'
 
       - name: Merge Test Reports
-        if: success() || failure()
+        if: always()
         run: npx junit-report-merger junit.xml "**/*.xml"
 
       - uses: actions/upload-artifact@v4
-        if: success() || failure()
+        if: always()
         with:
           name: test-results
           path: junit.xml
+
+      - name: Failed build detection
+        uses: actions/github-script@v7
+        if: >-
+          ${{
+               contains(needs.*.result, 'failure')
+            || contains(needs.*.result, 'cancelled')
+            || contains(needs.*.result, 'skipped')
```

Comment:
Do we think that skipped or cancelled tests are automatic test failures?

Reply:
Yes in this case yeah.
If not all tests complete successfully that means that we have an issue.